scipy/stats/_stats_py.py

```diff
@@ -4126,20 +4137,21 @@ def f_oneway(*samples, axis=0):
     # to a shift in location, and centering all data around zero vastly
     # improves numerical stability.
     offset = alldata.mean(axis=axis, keepdims=True)
-    alldata -= offset
+    alldata = alldata - offset
```

Comment:
Inplace operators don't work when working with different types. So, removed in place operations from `f_oneway`

Reply:
When do you get different types? Is it because you removed the conversion to `float`?

Reply:
Yes.